### ISSUE TITLE
SSL VHOSTS Possible?

### ISSUE THREAD

**Issue by infosecgeek** [securestate/king-phisher#117]
## Question

Apologies if this has been addressed elsewhere (I searched, promise) but, is there a way to add multiple SSL certs/vhosts to the server config in order to run multiple campaigns using SSL certs with separate domains? (https://<domain 1>, https://<domain 2>)

**Comment by zeroSteiner** [comment]
Unfortunately right now this is not possible. I'll have to look at the Python APIs to see if I can add it.

**Comment by zeroSteiner** [comment]
Ok, I've looked into this and got an SNI example working with Python. Looks like SNI has been supported since 2.7.9 and 3.4, so it's a relatively newer feature. Now this change specifically would need to be implemented in [AdvancedHTTPServer](https://github.com/zeroSteiner/AdvancedHTTPServer) which is what actually handles the server's SSL configureation etc. The good news is that it's also my project and I'm currently in the process of writing [version 2](https://github.com/zeroSteiner/AdvancedHTTPServer/tree/v2) and can add SNI to the list of things to do.

TLDR, I can get this added but it will take a while (couple months most likely).

**Comment by infosecgeek** [comment]
Fantastic, will definitely be welcome functionality when added. Thanks.

**Comment by zeroSteiner** [comment]
@infosecgeek just wanted to give you an update that this feature will be included in the v1.4 release. It's fully implemented and currently living in the dev branch where we do our testing before merging over into master. Once it's available you just need to specify `ssl_hosts` entries like in the [configuration template](https://github.com/securestate/king-phisher/blob/dev/data/server/king_phisher/server_config.yml#L83-87).

Thanks for this great feature request.
